Algebra.java

```diff
@@ -1,67 +1,144 @@
+		System.out.println(minus(2, 7)); // 2 - 7
+		System.out.println(times(3, 4)); // 3 * 4
+		System.out.println(plus(2, times(4, 2))); // 2 + 4 * 2
+		System.out.println(pow(5, 3)); // 5^3
+		System.out.println(pow(3, 5)); // 3^5
+		System.out.println(pow(-2, 2));
+		System.out.println(div(12, 3)); // 12 / 3
+		System.out.println(div(5, 5)); // 5 / 5
+		System.out.println(div(25, 7)); // 25 / 7
+		System.out.println(mod(25, 7)); // 25 % 7
+		System.out.println(mod(120, 6)); // 120 % 6
+		System.out.println(sqrt(36));
 		System.out.println(sqrt(263169));
-   		System.out.println(sqrt(76123));
-	}  
+		System.out.println(sqrt(76123));
+		System.out.println(pow(3, 5)); // 3^5
+		System.out.println(pow(-2, 5));
+	}
 
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int result = x1;
+		if (x2 > 0) {
+			for (int i = 0; i < x2; i++) { // x1 = x2 +x1
+				result++;
+			}
+		} else {
+			for (int i = 0; i > x2; i--) { // x1 = x1 + (-x2)
+				result--;
+			}
+		}
+
+		return result;
 	}
 
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int result = x1;
+		if (x2 > 0) {
+			for (int i = 0; i < x2; i++) { // x1 = x1 -x2
+				result--;
+			}
+		} else {
+			for (int i = 0; i > x2; i--) { // x1 = x1 -(-x2)
+				result++;
+			}
+		}
+		return result;
 	}
 
 	// Returns x1 * x2
 	public static int times(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int temp1 = x1, temp2 = x2;
+		if (x1 == 0 || x2 == 0) {
+			return 0;
+		}
+		if (x1 < 0) {
+			temp1 = minus(0, x1);
+		}
+		if (x2 < 0) {
+			temp2 = minus(0, x2);
+		}
+		int result = temp1;
+		for (int i = 1; i < temp2; i++) {
+			result = plus(result, temp1);
+		}
+		if (!((x1 > 0 && x2 > 0) || (x1 < 0 && x2 < 0))) {
+			result = minus(0, result);
+		}
+		return result;
 	}
 
 	// Returns x^n (for n >= 0)
 	public static int pow(int x, int n) {
-		// Replace the following statement with your code
-		return 0;
+		int resultpow = x;
+		if (n == 0) {
+			return 1;
+		}
+		for (int i = 1; i < n; i++) {
+			resultpow = times(resultpow, x);
+		}
+		return resultpow;
 	}
 
-	// Returns the integer part of x1 / x2 
+	// Returns the integer part of x1 / x2
 	public static int div(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int result = 1;
+		if (x1 == 0) {
+			return 0;
+		}
+		int temp1 = x1, temp2 = x2;
+		if (x1 < 0) {
+			temp1 = minus(0, x1);
+		}
+		if (x2 < 0) {
+			temp2 = minus(0, x2);
+		}
+		for (int i = 1; i < temp1; i++) {
+			if (times(temp2, i) == temp1) {
+				if ((x1 > 0 && x2 > 0) || (x1 < 0 && x2 < 0)) {
+					return i;
+				} else {
+					return minus(0, i);
+				}
+			} else if (times(temp2, i) > temp1) {
+				i--;
+				if ((x1 > 0 && x2 > 0) || (x1 < 0 && x2 < 0)) {
+					return i;
+				} else {
+					return minus(0, i);
+				}
+			}
```

Comment:
1. I highly recommend to not change the for loops variables inside the loop (see 115).
2. you have repeated code, it can be simplified.
consider the following: 
```
if (times(temp2, i) >= temp1) {
         int currentIndexIValueCalc = times(temp2, i) == temp1 ? i : minus(i,1);				
         if ((x1 > 0 && x2 > 0) || (x1 < 0 && x2 < 0)) {
                  return i;
         } else {
                  return -i;
         }
}
```